modules/mvt/src/lib/parse-mvt.js

```diff
@@ -1,22 +1,37 @@
 // import {VectorTile} from '@mapbox/vector-tile';
 import VectorTile from './mapbox-vector-tile/vector-tile';
+import BinaryVectorTile from './binary-vector-tile/vector-tile';
 
-import {geojsonToBinary} from '@loaders.gl/gis';
+import {featuresToBinary} from './binary-vector-tile/features-to-binary';
 import Protobuf from 'pbf';
 
 /*
   * Parse MVT arrayBuffer and return GeoJSON.
   *
   * @param {arrayBuffer} _ A MVT arrayBuffer
-  * @return {?Object} A GeoJSON geometry object
+  * @return {?Object} A GeoJSON geometry object or a binary representation
   */
 export default function parseMVT(arrayBuffer, options) {
   options = normalizeOptions(options);
 
+  const binary = options.gis.format === 'binary';
   const features = [];
+  const firstPassData = {
```

Comment:
One idea to generalize this would be something like a `BinaryFeatureWriter` class in `@loaders.gl/gis`:

```js
const writer = new BinaryFeatureWriter();

writer.addPoint(coordinates, properties);
writer.addLineString(coordinates, properties);
writer.addPolygon(coordinates, properties);

...

return writer.getData();
```

The class can allocate the typed arrays dynamically (arbitrary initial size; double size when overflown) without knowing how many features/vertices there will be.

This way, you don't have to modify the feature parser to get statistics, and you don't have to hold on to all features in memory.

Reply:
I think this could be interesting to explore in the future, but I would say out of scope for this PR.